### ISSUE TITLE
AIX compiler failure for JDK8 & 11

### ISSUE THREAD

**Issue by M-Davies** [eclipse/openj9#9369]
Possibly related to https://github.com/eclipse/openj9/issues/8508

https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-openj9
https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-aix-ppc64-openj9
```
Running ddrgen to generate j9ddr.dat and superset.dat
Error: dwarf/AixSymbolTableParser.cpp:698 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: ';'
Error: dwarf/AixSymbolTableParser.cpp:698 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: ';'
run_omrddrgen.mk:82: recipe for target '../j9ddr.dat' failed
gmake[2]: *** [../j9ddr.dat] IOT/Abort trap (core dumped)
/ramdisk0/build/workspace/build-scripts/jobs/jdk8u/jdk8u-aix-ppc64-openj9/workspace/build/src/closed/CopyToBuildJdk.gmk:241: recipe for target 'run-ddrgen' failed
gmake[1]: *** [run-ddrgen] Error 2
/ramdisk0/build/workspace/build-scripts/jobs/jdk8u/jdk8u-aix-ppc64-openj9/workspace/build/src/closed/make/Main.gmk:38: recipe for target 'build-j9vm' failed
gmake: *** [build-j9vm] Error 2
```
Failing since 22/04/2020

**Comment by pshipton** [comment]
What changed at Adopt? The builds at OpenJ9 are fine.
https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK8_ppc64_aix/1592/
https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK11_ppc64_aix/1612/
@sxa @keithc-ca

**Comment by keithc-ca** [comment]
We had similar failures until we realized that `export LDR_CNTRL=MAXDATA=0x80000000` was necessary (the process is just running out of memory).

**Comment by sxa** [comment]
We've always had that at adopt: https://github.com/AdoptOpenJDK/openjdk-build/blob/master/build-farm/platform-specific-configurations/aix.sh#L44

**Comment by M-Davies** [comment]
From https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-openj9/635/consoleFull...
```
Using /ramdisk0/build/tmp for temporary files (Clearing it out first...)
Found 4 items in it - removing them

real	0m0.273s
user	0m0.081s
sys	0m0.077s
LDR_CNTRL=MAXDATA=0x80000000
```

**Comment by pshipton** [comment]
From Slack:
Stewart X Addison 
The build of the openj9-0.20.0 stream that I did as a test was ok and not subject to https://github.com/eclipse/openj9/issues/9369 - so whatever that problem is it only seems to be from builds from the development branches starting from April 22nd.

The last successful build.
```
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-202004212129-b09)
Eclipse OpenJ9 VM (build master-be6b41b, JRE 1.8.0 AIX ppc64-64-Bit Compressed References 20200421_629 (JIT enabled, AOT enabled)
OpenJ9   - be6b41b
OMR      - eb6b5c5
JCL      - 5197128 based on jdk8u252-b09)
```

The Adopt builds don't seem to print the SHAs it's building from.
Below are the changes from the last successful until April 23, but I don't see anything obvious.
https://github.com/eclipse/openj9/compare/be6b41b...481d5c3
https://github.com/eclipse/openj9-omr/compare/eb6b5c5...31fb80f
https://github.com/ibmruntimes/openj9-openjdk-jdk8/compare/5197128...980f351

**Comment by sxa** [comment]
A few points:
- It doesn't seem to affect JDK14
- The builds started failing on the same day OpenSSL was bumped to 1.1.1.g
- The OpenJ9 0.20.0 release branch (also configured to pull OpenSSL 1.1.1g) is not affected

**Comment by pshipton** [comment]
It may be worthwhile rebuilding the Apr 22 build that worked to see if it still works. Otherwise, doing more custom builds and backing off the SHAs to discover which change caused the failure.

**Comment by sxa** [comment]
Given that the openj9-0.20.0 branch is ok I would be quite surprised if the older nightlies wouldn't be ok too. But regardless, won't `get_source.sh` in the extensions repo always pull the latest code from OpenJ9 and OMR instead of the ones built at that time? I guess it will be a bit of manual work if we want to do that test.

**Comment by pshipton** [comment]
> won't get_source.sh in the extensions repo always pull the latest code

Yes it will. Although it also takes options (i.e. `-openj9-sha` so the SHAs can be specified.

**Comment by M-Davies** [comment]
[Today's build](https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-openj9/639/) passed on `build-osuosl-ppc64-aix-71-2`

**Comment by sxa** [comment]
@M-Davies The link you've give is to a release build from the release branch ([for this requirement](https://github.com/AdoptOpenJDK/TSC/issues/155#issuecomment-620445417)), not a nightly therefore expected to pass as per my bullet point #3 above

**Comment by sxa** [comment]
We currently restrict the RAM limit given to the build for JDK14/J9 (not for HotSpot or earlier JDK versions) as per https://github.com/AdoptOpenJDK/openjdk-build/blob/master/build-farm/platform-specific-configurations/aix.sh#L110

It may be that we need to start restricting it for all OpenJ9 variants now because the RAM requirements during the build have increased (Currently building with 9 cores/10Gb of RAM based on the `configure` output). I'll need to do some testing to verify if that is the problem.

**Comment by sxa** [comment]
Reducing the RAM allowed to it (configure setting only one core to use) has made no difference

**Comment by sxa** [comment]
This is quite odd. I've been trying various builds including one with SHAs from around the before it started failing and this has the same crash in it.

extensions:
```
980f351a43f2fd13fa447a2cc578b7f159e7cbb2 (Apr 22 - Notarization for macos)
```
openj9:
```
de8a28d2ce6b42cd672c99b5251827b3e08167ba (Apr21 - doc update)
```
omr:
```
de8a28d2ce6b42cd672c99b5251827b3e08167ba (April 21 - doc update)
```
It's worth pointing out that the error lines with `Unknown stabs tag` occur in the successful bulids (from the GA branch) too.

I have a couple of options that I can see:
1. Identify the failing command and run it in isolation outside the full build process
2. Start building the extensions repository with OpenJ9/OMR SHAs from the GA branch

**Comment by lumpfish** [comment]
In the last good build, the same program throws an errors but it does not cause a crash:
Last good build: https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-openj9/639/consoleFull
```
Running ddrgen to generate j9ddr.dat and superset.dat
Error: dwarf/AixSymbolTableParser.cpp:700 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: 'vu7[f:identityHashDataAddRange__19MM_GCExtensionsBaseFP18MM_EnvironmentBaseP17MM_MemorySubSpaceUlPvT4:2890;vu8[f:identityHashDataRemoveRange__19MM_GCExtensionsBaseFP18MM_EnvironmentBaseP17MM_MemorySubSpaceUlPvT4:2890;vo4[f:computeDefaultMaxHeap__19MM_GCExtensionsBaseFP18MM_EnvironmentBase:2890;vu6[fi:setTenureAddressRange__19MM_GCExtensionsBaseFPvUl:2890;;'
Error: dwarf/AixSymbolTableParser.cpp:700 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: 'vu7[f:identityHashDataAddRange__19MM_GCExtensionsBaseFP18MM_EnvironmentBaseP17MM_MemorySubSpaceUlPvT4:2890;vu8[f:identityHashDataRemoveRange__19MM_GCExtensionsBaseFP18MM_EnvironmentBaseP17MM_MemorySubSpaceUlPvT4:2890;vo4[f:computeDefaultMaxHeap__19MM_GCExtensionsBaseFP18MM_EnvironmentBase:2890;vu6[fi:setTenureAddressRange__19MM_GCExtensionsBaseFPvUl:2890;;'
Blob written to file: ../j9ddr.dat
```
Latest nightly build: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-aix-ppc64-openj9/644/console
```
Running ddrgen to generate j9ddr.dat and superset.dat
Error: dwarf/AixSymbolTableParser.cpp:698 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: ';'
Error: dwarf/AixSymbolTableParser.cpp:698 parseStabstringDeclarationIntoDwarfDie(const string &, Dwarf_Error *) - Unknown stabs tag: ';'
run_omrddrgen.mk:82: recipe for target '../j9ddr.dat' failed
gmake[2]: *** [../j9ddr.dat] IOT/Abort trap (core dumped)
```
So maybe in the first case there is a tag value which is not liked, whereas in the second case the tag is missing?
Where is the tag supposed to come from?

**Comment by sxa** [comment]
Last good build is the GA one so was taken from those branches. We know that the GA branches and JDK14 nightlies are unaffected by this problem.

The command that's failing is:
`LIBPATH=.. ../ddrgen --blob ../j9ddr.dat --superset ../superset.dat --blacklist blacklist --macrolist ../macroList --overrides overrides --show-empty ../libj9ddr_misc29.debuginfo ../libj9gc29.debuginfo ../libj9jvmti29.debuginfo ../libj9prt29.debuginfo ../libj9shr29.debuginfo ../libj9thr29.debuginfo ../libj9trc29.debuginfo ../libj9vm29.debuginfo ../libj9vrb29.debuginfo ../libjclse29.debuginfo`
and the error shows if I run that at the command line outside the build process. (If I used `.so` files instead of `.debuginfo` ones then it completes successfully)

(Command above ran from `build/aix-ppc64-normal-server-release/vm/ddr`)

Size of the files referenced above are as follows:
```
bash-4.3$ ls -ld ../libj9ddr_misc29.debuginfo ../libj9gc29.debuginfo ../libj9jvmti29.debuginfo ../libj9prt29.debuginfo ../libj9shr29.debuginfo ../libj9thr29.debuginfo ../libj9trc29.debuginfo ../libj9vm29.debuginfo ../libj9vrb29.debuginfo ../libjclse29.debuginfo
-rwxr-xr-x    1 jenkins  staff        807346 May 04 04:17 ../libj9ddr_misc29.debuginfo
-rwxr-xr-x    1 jenkins  staff      78453681 May 04 04:17 ../libj9gc29.debuginfo
-rwxr-xr-x    1 jenkins  staff       5897056 May 04 04:16 ../libj9jvmti29.debuginfo
-rwxr-xr-x    1 jenkins  staff       3464638 May 04 03:43 ../libj9prt29.debuginfo
-rwxr-xr-x    1 jenkins  staff       7803454 May 04 03:50 ../libj9shr29.debuginfo
-rwxr-xr-x    1 jenkins  staff        460011 May 04 03:42 ../libj9thr29.debuginfo
-rwxr-xr-x    1 jenkins  staff       2264972 May 04 04:15 ../libj9trc29.debuginfo
-rwxr-xr-x    1 jenkins  staff      19682684 May 04 03:50 ../libj9vm29.debuginfo
-rwxr-xr-x    1 jenkins  staff      27396745 May 04 04:13 ../libj9vrb29.debuginfo
-rwxr-xr-x    1 jenkins  staff       7990239 May 04 04:16 ../libjclse29.debuginfo
bash-4.3$ 
```

**Comment by keithc-ca** [comment]
The 'unknown tag' message is because stabs strings are so long that they must be continued in (at least) a second segment. The version of `dump` in use truncates long stabs strings (before the signal that the string is continued). The continuation (that can't be recognized as such) will probably not be structurally valid leading to the diagnostic from `ddrgen`. Until now, it has been safe to just ignore the continuations. Perhaps the time has come to upgrade AIX (or at least the version of `dump` in use) to include a fix for this issue.

**Comment by sxa** [comment]
@keithc-ca Do you know what changed in the codebase that has triggered the new crashes and why 14 is immune to it?

**Comment by keithc-ca** [comment]
No, I don't know what changed, specifically. Many data structures have portions that are configuration-dependent or version-dependent; descriptions of those data structures as stabs strings will of course vary by configuration, version and time. We have probably just been lucky in the past (or are still lucky in the case of 14). I believe IBM internal machines have the newer (fixed) version of `dump` installed which may explain why we don't see this problem there.

**Comment by sxa** [comment]
If we have no way of working around this then we'll need to reopen https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1006 or create a new requirement for an alternate 7.1.SP5TL5 build machine to be used for building OpenJ9 at adoptopenjdk

**Comment by sxa** [comment]
I'm going to send all the builds tonight onto an AIX7.1.5 machine, but it's one that's being used for test at the moment so we'll see how it goes tonight and then decide what to do

**Comment by sxa** [comment]
@jdekonin What bash version do you have on your AIX 715 systems? I'm getting this error on mine during the build:
```
/opt/freeware/bin/bash: cannot make pipe for process substitution: No such file or directory
/opt/freeware/bin/bash: /tmp//sh-np.p83aac: ambiguous redirect
```
My version is `GNU bash, version 5.0.0(2)-release (powerpc-ibm-aix6.1.9.0)`

Cc @sej-jackson

**Comment by sxa** [comment]
Also @jdekonin which boot JDK do you have on those systems for JDK8? The one I've got on my new 715 system doesn't appear to work, so I've switched it to a JDK8 for now...)

**Comment by jdekonin** [comment]
The bootjdk is the same as provided via playbook afaik, bash is the latest available through the yum installer.  From a yum perspective the only packages pending update are db to v6.2.38-1 and git to v2.20.2-1, so pretty current.

```
# oslevel -s
7100-05-05-1939
-bash-5.0$ /opt/java7/bin/java -version
openjdk version "1.7.0-internal"
OpenJDK Runtime Environment (build 1.7.0-internal-jvmtests_2017_05_20_22_30-b00)
OpenJDK 64-Bit Server VM (build 24.95-b00, mixed mode)
-bash-5.0$ bash --version
GNU bash, version 5.0.0(2)-release (powerpc-ibm-aix6.1.9.0)
```

**Comment by jdekonin** [comment]
Reading further this is a dup of the issue I opened back in Feb.
https://github.com/eclipse/openj9/issues/8508

I found a ulimit increase resolved the problem but was unable to get anyone to comment on what the defaults should be for the users.  I don't believe setting ulimit settings to unlimited is valid.

**Comment by sxa** [comment]
I'll take a look at that - thanks. I had seen it beore but it wasn't a blocker at the time. For the record, if I replace `bash` on the machine with an earlier release (`GNU bash, version 4.3.30(1)-release (powerpc-ibm-aix6.1.0.0)
` - I copied the binary from another system) it also runs through ok without the error.

**Comment by sxa** [comment]
I still haven't been able to resolve the issue with the bash going pop with the `cannot make pipe` error, so for now I've left it running the earlier version that I've copied across. Need to understand why this isn't seen on the OpenJ9 systems which appear to be running the same version of bash (although potentially not of other libraries). The pipe ulimit of 64 seems the same on both systems
